pkg/services/object/put/service_test.go

```diff
@@ -0,0 +1,642 @@
+	}
+	commonPrm, err := objutil.CommonPrmFromRequest(localReq)
+	if err != nil {
+		panic(err)
+	}
+
+	var ip PutInitPrm
+	ip.WithObject(hdr.CutPayload())
+	ip.WithCommonPrm(commonPrm)
+
+	if err := stream.Init(&ip); err != nil {
+		return nil, err
+	}
+
+	return (*testPayloadStream)(stream), nil
+}
+
+func (m *serviceClient) ReplicateObject(context.Context, oid.ID, io.ReadSeeker, neofscrypto.Signer, bool) (*neofscrypto.Signature, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectDelete(context.Context, cid.ID, oid.ID, user.Signer, client.PrmObjectDelete) (oid.ID, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectGetInit(context.Context, cid.ID, oid.ID, user.Signer, client.PrmObjectGet) (object.Object, *client.PayloadReader, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectHead(context.Context, cid.ID, oid.ID, user.Signer, client.PrmObjectHead) (*object.Object, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectSearchInit(context.Context, cid.ID, user.Signer, client.PrmObjectSearch) (*client.ObjectListReader, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) SearchObjects(context.Context, cid.ID, object.SearchFilters, []string, string, neofscrypto.Signer, client.SearchObjectsOptions) ([]client.SearchResultItem, string, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectRangeInit(context.Context, cid.ID, oid.ID, uint64, uint64, user.Signer, client.PrmObjectRange) (*client.ObjectRangeReader, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ObjectHash(context.Context, cid.ID, oid.ID, user.Signer, client.PrmObjectHash) ([][]byte, error) {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) AnnounceLocalTrust(context.Context, uint64, []apireputation.Trust, client.PrmAnnounceLocalTrust) error {
+	// TODO: interfaces are oversaturated. This will never be needed to server object PUT. Refactor this.
+	panic("unimplemented")
+}
+
+func (m *serviceClient) AnnounceIntermediateTrust(context.Context, uint64, apireputation.PeerToPeerTrust, client.PrmAnnounceIntermediateTrust) error {
+	panic("unimplemented")
+}
+
+func (m *serviceClient) ForEachGRPCConn(context.Context, func(context.Context, *grpc.ClientConn) error) error {
+	panic("unimplemented")
+}
+
+type testPayloadStream Streamer
+
+func (x *testPayloadStream) Write(p []byte) (int, error) {
+	if err := (*Streamer)(x).SendChunk(new(PutChunkPrm).WithChunk(p)); err != nil {
+		return 0, err
+	}
+	return len(p), nil
+}
+
+func (x *testPayloadStream) Close() error {
+	_, err := (*Streamer)(x).Close()
+	return err
+}
+
+func (testPayloadStream) GetResult() client.ResObjectPut {
+	return client.ResObjectPut{}
+}
+
+type testCluster struct {
+	nodeServices      nodeServices
+	nodeNetworks      []mockNetwork
+	nodeSessions      []mockNodeSession
+	nodeLocalStorages []inMemLocalStorage
+}
+
+func (x testCluster) allStoredObjects() iter.Seq[[]object.Object] {
```

Comment:
Looks like you're never iterating it partially, so `iter.Seq` doesn't buy us anything here.

Reply:
changed to slice